napari/_vispy/vispy_base_layer.py

```diff
@@ -127,21 +127,25 @@ def _on_blending_change(self, event=None):
         self.node.update()
 
     def _on_scale_change(self, event=None):
-        self.scale = [
-            self.layer.scale[d] * self.layer._scale_view[d]
-            for d in self.layer.dims.displayed[::-1]
-        ]
+        scale = (
+            self.layer._transform_view.compose(self.layer._transform)
+            .set_slice(self.layer.dims.displayed)
+            .scale
+        )
+        self.scale = scale[::-1]
```

Comment:
Can you add a comment here, "convert NumPy axis ordering to VisPy axis ordering"? Same below.

Reply:
done